bin/go-rpm-integration

```diff
@@ -262,9 +267,8 @@ case $action in
                       ln -fs "${sourcedir}"           "${GO_BUILD_PATH}/src/${goipath}"
                     fi
                     pushd "${GO_BUILD_PATH}/src/${goipath}" >/dev/null
-                      [[ ! -e .goipath ]] && touch .goipath
-                      golistfiles=$(listfiles "${goipath}")
-                      for file in ${golistfiles[@]} ${otherfiles[@]} .goipath ; do
+                      echo "${golistflags[$goipath]}" > .goipath
```

Comment:
If you want something written into the `.goipath` file, it needs to be something describing the import path prefix itself.

Though, passing flags does not seem to be the right data IMHO.

Reply:
Feel free to change it to .goflags if you want

Recording flags in a dotfile is the only reliable way to pass them to the correct subpackage, without doing complex muxing-demuxing that does not really work in all cases

This way of doing things is simple, transparent and robust

Reply:
`.goflags` sounds definitely better. Please use `.goflags` + rebase. I will merge then.

Reply:
Alternatively, to avoid multiplying .files, .goipath can be turned into a full manifest with more attributes added over time as we need them.

However that requires defining a syntax that is not to hard to generate in shell and macros (.ini, yaml, whatever), and adding the corresponding syntax parser to golist

Reply:
I would not want to go that way unless necessary. If we decide to do that it would make more sense to write a new tooling (purely in Go) that will actually run on top of the rpm. Let's stick with the lockfiles for now.

What you are suggesting requires broader discussion. And we definitely need more folks with experience to help us make the right decisions.

Reply:
Whatever we end up doing, forget about the “new pure tooling on top of rpm”.  The whole idea is dead dead beaten to death horse no use trying to relive it one more time.

It has been tried countless times by people more interested in language purity or experimenting than in getting working production solutions. They always try to “simplify” the needs to avoid dealing with hard problems and by the time they realise they can't avoid dealing with the hard problems they usually declare the experiment a success and move on to something else to avoid admitting their design is a failure. The strength of rpm is that it is on the service of actual long lived distributions people depend on, it forces its maintainers to be pragmatic and deal with hard stuff without postponing them to some illusory future.

Demonstrating something able to deploy simplified software patterns once is trivial. Dealing with real-world software, continuously updated over time, that get renamed renumbered split merged forked reconfigured is several orders of magnitudes harder. The competitors got closer to rpm over time not the reverse, windows updates gets more like rpm every year and it comes from a completely different and segregated software culture. (it could write deb instead of rpm, real world constrains have forced a complete design convergence over time).

In the end of the day the rpm design is sound and time-tested and you always have apps that cross language barriers (case in point I got into this adventure because the prometheus snmp exporter needs to link to libsnmp and upstream is too enamoured with pure Go solutions to be able to publish any working binary for Fedora/EL7).

So what works: extending rpm integration to new languages with language-specific helpers. (rpm *can* be extended and adapted to changes. Purists always forget extension or change capabilities because they want the world to conform to their view, not adapt themselves to the world)

What does not work: restarting from scratch, expecting software ecosystem to follow a single, pure, mono-language design that makes rpm constrains unnecessary.